crawl-ref/source/spl-util.cc

```diff
@@ -1344,8 +1344,14 @@ int spell_highlight_by_utility(spell_type spell, int default_colour,
     {
         return COL_FORBIDDEN;
     }
+    // Grey out spells for which you lack experience or spell levels.
+    if (spell_difficulty(spell) > you.experience_level
+             || player_spell_levels() < spell_levels_required(spell))
```

Comment:
another copy/paste thing after I floated these lines to the new function. I'll tweak it